sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncItemResponse.java

```diff
@@ -51,21 +57,99 @@ public T getResource(){
-                        super.setProperties(null);
-                    } else {
-                        CosmosItemProperties props = new CosmosItemProperties(responseBodyString);
-                        super.setProperties(props);
-                    }
-                }
+        synchronized (this) {
+            if (StringUtils.isEmpty(responseBodyString)) {
+                props = null;
+            } else {
+                props = new CosmosItemProperties(responseBodyString);
             }
+
         }
     }
+
+    /**
+     * Gets the maximum size limit for this entity (in megabytes (MB) for server resources and in count for master
+     * resources).
+     *
+     * @return the max resource quota.
+     */
+    public String getMaxResourceQuota() {
+        return resourceResponse.getMaxResourceQuota();
+    }
+
+    /**
+     * Gets the current size of this entity (in megabytes (MB) for server resources and in count for master resources)
+     *
+     * @return the current resource quota usage.
+     */
+    public String getCurrentResourceQuotaUsage() {
+        return resourceResponse.getCurrentResourceQuotaUsage();
+    }
+
+    /**
+     * Gets the Activity ID for the request.
+     *
+     * @return the activity getId.
+     */
+    public String getActivityId() {
+        return resourceResponse.getActivityId();
+    }
+
+    /**
+     * Gets the number of index paths (terms) generated by the operation.
+     *
+     * @return the request charge.
+     */
+    public double getRequestCharge() {
+        return resourceResponse.getRequestCharge();
+    }
+
+    /**
+     * Gets the HTTP status code associated with the response.
+     *
+     * @return the status code.
+     */
+    public int getStatusCode() {
+        return resourceResponse.getStatusCode();
+    }
+
+    /**
+     * Gets the token used for managing client's consistency requirements.
+     *
+     * @return the session token.
+     */
+    public String getSessionToken() {
+        return resourceResponse.getSessionToken();
+    }
+
+    /**
+     * Gets the headers associated with the response.
+     *
+     * @return the response headers.
+     */
+    public Map<String, String> getResponseHeaders() {
+        return resourceResponse.getResponseHeaders();
+    }
+
+    /**
+     * Gets the diagnostics information for the current request to Azure Cosmos DB service.
+     *
+     * @return diagnostics information for the current request to Azure Cosmos DB service.
+     */
+    public CosmosResponseDiagnostics getCosmosResponseDiagnostics() {
+        return resourceResponse.getCosmosResponseDiagnostics();
+    }
+
+    /**
+     * Gets the end-to-end request latency for the current request to Azure Cosmos DB service.
+     *
+     * @return end-to-end request latency for the current request to Azure Cosmos DB service.
+     */
+    public Duration getRequestLatency() {
+        return resourceResponse.getRequestLatency();
```

Comment:
This is already available in cosmosResponseDiagnostics exposed above. duplicate ways of getting the same thing